test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java

```diff
@@ -144,4 +144,9 @@ public interface UniAsserter {
      * Clear the test data.
      */
     void clearData();
+
+    /**
+     * @return a {@link Uni} representing the operations pipeline up to this point
+     */
+    Uni<?> asUni();
```

Comment:
I'm not 100% sure it's a good idea to expose the current execution directly as it opens the door for "creative usage" of the API. That was the reason why the `DefaultUniAsserter` was used internally. Maybe we could introduce `AbstractUniAsserter` or something like that. But maybe I just overthink this...

Reply:
I got the reasoning behind the original approach. I've decided to use the asUni because it was the fastest way to check if the fix would work. Then I lost some time figuring out how to test it and forgot about it.

I'll see if I can introduce the abstract class.

Reply:
What about this: 3c00df9c644b0114e773315d6bf6dfa2b2e54ac5

Reply:
Better, except that I don't understand why the name `UnwrappableAsserter`? ;-)

Reply:
> Never question a name without suggesting a better one ;)

Ah, sorry about that ;-)

> But the idea is that the uni is wrapped by the class and this interface provide a way to unwrap it.

And do we really need an interface for this? I mean it could be a package-private abstract class or something like that...

Reply:
For sure, but I don't think we need this flexibility at the moment... it should be a small implementation detail, not a part of the public API at all.

Reply:
I've changed the scope to `package-private`. 
And the name to `UnwrappableUniAsserter`.

I don't plan to do any extra work on this issue (if it passes all the checks on CI).
I don't see any drawback of having an interface in this situation. But feel free to change it any way you prefer.

Reply:
No problem. It's perfect. Thanks.